### PR TITLE
Implement xlink mappings for name and alternativeName.

### LINKED ISSUE #2102: Implement xlink mappings for name and alternativeName
See https://github.com/sul-dlss/dor-services-app/blob/main/spec/services/cocina/mapping/descriptive/mods/name_spec.rb

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2102

## Why was this change made?
Mapping goodness.

## How was this change tested?
Unit, sdr-deploy

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

**Comment by justinlittman** [comment]
Before:
```
Status (n=100000):
  Success:   98313 (98.313%)
  Different: 1156 (1.156%)
  To Cocina error:     0 (0.0%)
  To Fedora error:     0 (0.0%)
  Missing:     531 (0.531%)
```

After:
```
Status (n=100000):
  Success:   98313 (98.313%)
  Different: 1156 (1.156%)
  To Cocina error:     0 (0.0%)
  To Fedora error:     0 (0.0%)
  Missing:     531 (0.531%)
```

File: spec/services/cocina/mapping/descriptive/mods/name_spec.rb [hunk 1]
```diff
@@ -1481,74 +1481,81 @@
...
+      let(:cocina) do
+        {
+          contributor: [
+            {
+              name: [
+                {
+                  groupedValue: [
+                    {
+                      value: 'Strachey, Dorothy',
+                      # TOOD: verify
```
Comment (jcoyne) [inline spec/services/cocina/mapping/descriptive/mods/name_spec.rb:1542]: Typo 😆

**Review by jcoyne — APPROVED** [review]
Looks good. It would be nice to have a ticket for that "TOOD"

File: spec/services/cocina/mapping/descriptive/mods/name_spec.rb (hunk 1 above)
Comment (justinlittman) [reply spec/services/cocina/mapping/descriptive/mods/name_spec.rb:1542]: Cruft. Removing.

**Pull request merged**